app/controllers/parole_reviews_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class ParoleReviewsController < PrisonsApplicationController
+  before_action :load_offender
+
+  def edit
+    @parole_review = ParoleReview.find_by(review_id: params[:id])
+  end
+
+  def update
+    @parole_review = ParoleReview.find_by(review_id: params[:id])
+    hearing_outcome_received_date = validate_date(params)
+    if @parole_review.errors.empty? && @parole_review.update(hearing_outcome_received_on: hearing_outcome_received_date)
+      RecalculateHandoverDateJob.perform_now(@offender.offender_no)
+      redirect_to prison_prisoner_path(prison: @prison, id: @offender.offender_no)
+    else
+      render 'edit'
+    end
+  end
+
+private
+
+  def load_offender
+    @offender = OffenderService.get_offender(params[:prisoner_id])
+  end
+
+  def validate_date(params)
```

Comment:
If you moved this to the model you could remove a lot of manual logic

```
class ParoleReview
  validates :hearing_outcome_received_on, presence: { message: 'Enter the full date the hearing outcome was confirmed' }
  validate :hearing_outcome_received_on_must_be_in_past

  def hearing_outcome_received_on_must_be_in_past
    if hearing_outcome_received_on.present? && hearing_outcome_received_on.past?
      errors.add(:hearing_outcome_received_on, "The date the hearing outcome was confirmed must be in the past")
    end
  end
```

As the model knows how to take in each of the date part params.

Reply:
You might want conditional validation https://guides.rubyonrails.org/active_record_validations.html#on